mne/stats/cluster_level.py

```diff
@@ -513,15 +514,15 @@ def _find_clusters_1dir(x, x_in, connectivity, max_step, t_power, ndimage):
         else:
             # boolean masks (raveled)
             clusters = list()
-            sums = np.empty(n_labels)
-            for label in range(n_labels):
-                c = labels == label + 1
+            sums = list()
+            for label in range(1, n_labels + 1):
+                c = labels == label
                 clusters.append(c.ravel())
                 if t_power == 1:
-                    sums[label] = np.sum(x[c])
+                    sums.append(np.sum(x[c]))
                 else:
-                    sums[label] = np.sum(np.sign(x[c]) *
-                                         np.abs(x[c]) ** t_power)
+                    sums.append(np.sum(np.sign(x[c]) *
+                                       np.abs(x[c]) ** t_power))
```

Comment:
This ends up being less efficient than before, because now you'll end up with two copies of `sums` -- the first in a list and then another made later with `atleast_1d`.

In general we tend to preallocate and assign when weknow the size we will have (as is done here in `master`) and reserve list-append-convert-ndarray for cases where we don't know the size ahead of time (as is the case above, and why we use lists there), so conceptually this goes against that a bit, too...

Reply:
Thanks for explaining! I will change it back